app/uk/gov/hmrc/gform/gform/csv/CsvTaxRateAdapter.scala

```diff
@@ -33,36 +38,33 @@ class CsvTaxRateAdapter extends CsvDataRetrieveAdapter[TaxRate] {
       TaxRate(
         row("TaxRegime"),
         row("Code"),
-        row("Rate"),
+        BigDecimal(row("Rate")).setScale(2),
         LocalDate.parse(row("StartDate"), hmrcTaxRateDateFormat),
         LocalDate.parse(row("EndDate"), hmrcTaxRateDateFormat)
       )
     }
 
-  override def search(request: DataRetrieve.Request): Option[JsValue] = {
-    val taxRateRequest: TaxRateRequest = request.json.as[TaxRateRequest]
-    val regimeRates: Option[List[TaxRate]] = data.groupBy(_.regime).get(taxRateRequest.regime)
-    val requestedDate: LocalDate = LocalDate.parse(taxRateRequest.date, hmrcTaxRateRequestFormat)
-
-    val searchResult: Option[List[TaxRate]] = regimeRates.map(_.filter { rate =>
-      rate.code == taxRateRequest.code && (
-        rate.startDate.isEqual(requestedDate) || rate.startDate.isBefore(requestedDate)
-      ) && (
-        rate.endDate.isEqual(requestedDate) || rate.endDate.isAfter(requestedDate)
-      )
-    })
+  override def search(request: DataRetrieve.Request): Option[JsValue] =
+    Try(request.json.as[TaxRateRequest]) match {
+      case Success(taxRateRequest: TaxRateRequest) =>
+        val regimeRates: Option[List[TaxRate]] = data.groupBy(_.regime).get(taxRateRequest.regime)
+        val requestedDate: LocalDate = LocalDate.parse(taxRateRequest.date, hmrcTaxRateRequestFormat)
 
-    // There must be only one...
-    val response: Option[TaxRate] = searchResult match {
-      case Some(list) if list.size == 1 => Some(list.head)
-      case _                            => None
+        val response: Option[TaxRate] = regimeRates.flatMap(_.find { rate =>
+          rate.code === taxRateRequest.code && (
+            rate.startDate.isEqual(requestedDate) || rate.startDate.isBefore(requestedDate)
+          ) && (
+            rate.endDate.isEqual(requestedDate) || rate.endDate.isAfter(requestedDate)
+          )
+        })
+        response.map(rate => Json.toJson(rate))
+      case Failure(exception) =>
+        logger.error(s"An error occurred attempting to unmarshal request: ${request.json}")
+        throw exception
```

Comment:
Perhaps better alternative would be to do this:
```
case JsError(e) =>
  throw new Exception(
    s"An error occurred attempting to unmarshal request: ${request.json}. Error: ${Json.stringify(JsError.toJson(e))}"
  )
```

I think explicit logging is not needed here. We can rely on default error handler logging instead.